src/receive_buffer.cpp

```diff
@@ -125,7 +125,7 @@ void receive_buffer::cut(int const size, int const packet_size, int const offset
 		m_recv_end -= size;
 
 #if TORRENT_USE_ASSERTS
-		std::fill(m_recv_buffer.begin() + m_recv_end, m_recv_buffer.end(), 0xcc);
+		std::fill(m_recv_buffer.begin() + m_recv_end, m_recv_buffer.end(), static_cast<unsigned char>(0xcc));
```

Comment:
as far as I can tell, ``m_recv_buffer.begin()`` returns a ``char*``, not ``unsigned char``. Am I missing something?